### PR TITLE
git: when importing refs, abandon commits that were abandoned in git

### PR DESCRIPTION
**Pull request by martinvonz**
# Checklist

- [x] I have made relevant updates to `CHANGELOG.md`

### CODE REVIEW

File: lib/src/git.rs [hunk 1]
```diff
@@ -57,9 +57,16 @@ pub fn import_refs(
     git_repo: &git2::Repository,
 ) -> Result<(), GitImportError> {
     let store = mut_repo.store().clone();
-    let git_refs = git_repo.references()?;
     let mut existing_git_refs = mut_repo.view().git_refs().clone();
+    let mut old_git_heads = HashSet::new();
```
Comment (spectral54) [inline lib/src/git.rs:61]: Would it make sense to make a non-mut view of these?

```
let old_git_heads = existing_git_refs
    .values()
    .iter().map(|old_target| old_target.adds())
    .flatten()
    .collect();
```

File: lib/src/git.rs [hunk 2]
```diff
@@ -117,6 +121,24 @@ pub fn import_refs(
...
+    // in jj as well.
+    let old_git_heads = old_git_heads.into_iter().collect_vec();
+    let new_git_heads = new_git_heads.into_iter().collect_vec();
+    // We could use mut_repo.record_rewrites() here but we know we only need to care
+    // about abandoned commits for now. We may want to change this if we ever
+    // add a way of preserving change IDs across rewrites by `git` (e.g. by
+    // putting them in the commit message).
+    let abandoned_commits = mut_repo
+        .index()
+        .walk_revs(&old_git_heads, &new_git_heads)
```
Comment (spectral54) [inline lib/src/git.rs:135]: Is there a way of using named parameters in rust, or some canonical mechanism to specify the name of the parameter in a way that linters would check? Having this be something like .walk_revs(wanted=&old_git_heads, unwanted=&new_git_heads)  would have made this a lot more readable for me.

**Review by spectral54 — APPROVED** [review]

File: lib/src/git.rs (hunk 1 above)
Comment (martinvonz) [reply lib/src/git.rs]: Thanks, that's much better (could even drop the `.iter()` and use `flat_map()`)

File: lib/src/git.rs (hunk 2 above)
Comment (martinvonz) [reply lib/src/git.rs:133]: I agree, but I think we'll have to wait for https://github.com/rust-lang/rfcs/issues/323, if that ever happens. (My IDE shows the argument names if they don't match the variable names, but that doesn't help during review etc.)

**Pull request merged**